framework/core/src/Foundation/Concerns/InteractsWithLaravel.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation\Concerns;
+
+use Flarum\Locale\LocaleManager;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+
+/**
+ * Implementation of the Laravel Application Contract,
+ * for the sake of better integration with Laravel packages/ecosystem.
+ */
+trait InteractsWithLaravel
+{
+    protected array $terminatingCallbacks = [];
+    protected ?bool $isRunningInConsole = null;
+
+    public function terminating($callback): static
+    {
+        $this->terminatingCallbacks[] = $callback;
+
+        return $this;
+    }
+
+    public function terminate(): void
+    {
+        $index = 0;
+
+        while ($index < count($this->terminatingCallbacks)) {
+            $this->call($this->terminatingCallbacks[$index]);
+
+            $index++;
+        }
+    }
+
+    public function basePath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->base, $path);
+    }
+
+    public function publicPath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->public, $path);
+    }
+
+    public function storagePath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->storage, $path);
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function bootstrapPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'bootstrap'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function configPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->storage, 'config'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function databasePath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'database'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function langPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'lang'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function resourcePath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'resources'),
+            $path
+        );
+    }
+
+    public function environment(...$environments)
+    {
+        if (count($environments) > 0) {
+            $patterns = is_array($environments[0]) ? $environments[0] : $environments;
+
+            return Str::is($patterns, $this['env']);
+        }
+
+        return $this['env'];
+    }
+
+    public function runningInConsole(): bool
```

Comment:
Do we need to cache this?

Reply:
it's cached

Reply:
Sorry, poor wording on my end. I mean, is this expensive to compute / is there a reason why we're caching it?

Reply:
oh, no it's not 🙈